File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int current_num = 0;
        int while_i = 0;

        // override all duplicates
        for (int i = 0; i < nums.size(); i=while_i) {
            current_num = nums[i];
            while_i = ++i;
            while (nums[while_i] == current_num) {
                nums[while_i] = -101;
                while_i++;
            }
        }

        // shift all elements that aren't -101 to the start
        int unique_idx = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != -101) {
                nums[unique_idx] = nums[i];
                unique_idx++;
            }
        }
        
        return unique_idx;
    }
};

//PSEUDO
/*
    1. for every elem, replace all dupli w 0
    2. shift all non -101 elems to start
    3. keep track of number of elements
    4. return that when end reached
*/

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
            1. for every elem
            2. loop through all elems
            3. if i + j == target, return i, j
        """
        for i in range(len(nums)):
            for j in range(i +  1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        
        